core/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -387,6 +397,46 @@ public String delResource(final ServiceContext context, final String metadataUui
         return String.format("Unable to remove resource '%s'.", resourceId);
     }
 
+    @Override
+    public void copyResources(ServiceContext context, String sourceUuid, String targetUuid, MetadataResourceVisibility metadataResourceVisibility) throws Exception {
+        final int sourceMetadataId = canDownload(context, sourceUuid, metadataResourceVisibility, false);
```

Comment:
Cannot hard code the approval flag to false.  I believe this will cause duplicate metadata function to fail. Because it would be copying potentially from a source that is approved.

Reply:
I also believe that you need to use canEdit instead of canDownload

Reply:
For the "canDownload" part, I am not so sure because this code is copied/pasted from its getResource method which is part of the replacing procedure. Have to do a local test to find out if they are identical

Reply:
Hardcode removed with old and new approval flag accordingly. CanEdit tested locally and work good. Resolved overall.